src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -271,16 +264,23 @@ class JesBackend extends Backend with LazyLogging {
         taskOutput.name -> attemptedValue
       } toMap
 
-      lazy val stderrLength: BigInteger = JesConnection.storage.objectSize(GoogleCloudStoragePath(stderrJesOutput(backendCall.callGcsPath).gcs))
-
-      if (backendCall.call.failOnStderr && stderrLength.intValue > 0) {
-        FailedExecution(new Throwable(s"Workflow ${backendCall.workflowDescriptor.id}: stderr has length $stderrLength for command: $command"))
-      } else status match {
-        case Run.Success =>
+      lazy val stderrLength: BigInteger = JesConnection.storage.objectSize(GoogleCloudStoragePath(backendCall.stderrJesOutput.gcs))
+      lazy val tryReturnCode = Try(backendCall.downloadRcFile.trim.toInt)
+      lazy val workflowId = backendCall.workflowDescriptor.id
+      lazy val continueOnReturnCode = backendCall.call.continueOnReturnCode
+
+      status match {
+        case Run.Success if backendCall.call.failOnStderr && stderrLength.intValue > 0 =>
+          FailedExecution(new Throwable(s"Workflow $workflowId: stderr has length $stderrLength for command: $command"))
+        case Run.Success if tryReturnCode.isFailure =>
+          FailedExecution(new Throwable(s"Workflow $workflowId: failed to download or parse return code file", tryReturnCode.failed.get))
+        case Run.Success if continueOnReturnCode.continueFor(tryReturnCode.get) =>
           unwrapOutputValues(outputMappings, backendCall.workflowDescriptor) match {
             case Success(outputs) => SuccessfulExecution(outputs)
             case Failure(e) => FailedExecution(e)
           }
+        case Run.Success =>
+          FailedExecution(new Throwable(s"Workflow $workflowId: disallowed command return code: " + tryReturnCode.get))
```

Comment:
Yes, I had it that way originally but I thought we had a preference for doing it more like this?  I'd personally prefer to do it the way you suggest. :smile: